tests/test_processors.py

```diff
@@ -714,3 +729,339 @@ def test_py3_exception_no_traceback(self):
         e = ValueError()
 
         assert (e.__class__, e, None) == _figure_out_exc_info(e)
+
+
+class TestCallsiteParameterAdder:
+    parameter_strings = {
+        "pathname",
+        "filename",
+        "module",
+        "func_name",
+        "lineno",
+        "thread",
+        "thread_name",
+        "process",
+        "process_name",
+    }
+
+    _all_parameters = set(CallsiteParameter)
+
+    def test_all_parameters(self) -> None:
+        """
+        All callsite parameters are included in ``self.parameter_strings`` and
+        the dictionary returned by ``self.get_callsite_parameters`` contains
+        keys for all callsite parameters.
+        """
+        assert self.parameter_strings == {
+            member.value for member in self._all_parameters
+        }
+        assert self.parameter_strings == self.get_callsite_parameters().keys()
+
+    @pytest.mark.xfail(
+        reason=(
+            "CallsiteParameterAdder cannot "
+            "determine the callsite for async calls."
+        )
+    )
+    @pytest.mark.asyncio
+    async def test_async(self) -> None:
+        """
+        Callsite information for async invocations are correct.
+        """
+        try:
+            string_io = StringIO()
+
+            class StingIOLogger(structlog.PrintLogger):
+                def __init__(self):
+                    super().__init__(file=string_io)
+
+            processor = self.make_processor(None, ["concurrent", "threading"])
+            structlog.configure(
+                processors=[processor, JSONRenderer()],
+                logger_factory=StingIOLogger,
+                wrapper_class=structlog.stdlib.AsyncBoundLogger,
+                cache_logger_on_first_use=True,
+            )
+
+            logger = structlog.stdlib.get_logger()
+
+            callsite_params = self.get_callsite_parameters()
+            await logger.info("baz")
+
+            assert {"event": "baz", **callsite_params} == json.loads(
+                string_io.getvalue()
+            )
+
+        finally:
+            structlog.reset_defaults()
+
+    def test_additional_ignores(self, monkeypatch: pytest.MonkeyPatch) -> None:
+        """
+        Stack frames from modules with names that start with values in
+        `additional_ignores` are ignored when determining the callsite.
+        """
+        test_message = "test message"
+        additional_ignores = ["tests.utils"]
+        processor = self.make_processor(None, additional_ignores)
+        event_dict: EventDict = {"event": test_message}
+
+        # WARNING: The below three lines are sensitive to relative line numbers
+        # (i.e. the invocation of processor must be two lines after the
+        # invocation of get_callsite_parameters) and is order sensitive (i.e.
+        # monkeypatch.setattr must occur after get_callsite_parameters but
+        # before invocation of processor).
```

Comment:
this is giving me shivers but it should be fine 😅

Reply:
It is quite annoying to work with also, I bumped my head against it when trying to change to using `additional_frame`, not sure if there are alternatives but happy to change if there are other options.